apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-plugin/src/main/java/gov/cms/model/dsl/codegen/plugin/GenerateEntitiesFromDslMojo.java

```diff
@@ -1017,12 +1023,41 @@ TypeSpec createAdditionalFieldsInnerClass(
     return classBuilder.build();
   }
 
+  /**
+   * Generates an accessor method for fields of the same group to be returned together.
+   *
+   * @param mapping {@link MappingBean} for the entity with the new fields
+   * @return the list of {@link MethodSpec}
+   */
+  @VisibleForTesting
+  List<MethodSpec> createMethodSpecsForGroupedProperties(MappingBean mapping) {
+    List<MethodSpec> methodSpecs = new LinkedList<>();
+    Map<String, List<ColumnBean>> groupedColumns =
+        mapping.getTable().getColumns().stream()
+            .filter(columnBean -> columnBean.hasGroupName())
+            .collect(Collectors.groupingBy(ColumnBean::getGroupName));
+    for (Map.Entry<String, List<ColumnBean>> entry : groupedColumns.entrySet()) {
+      final var columnBean = entry.getValue().stream().findFirst().get();
+      final var propertyType =
+          columnBean.isNullable()
+              ? ParameterizedTypeName.get(OptionalClassName, columnBean.computeJavaAccessorType())
+              : columnBean.computeJavaAccessorType();
+      final var methodSpec =
+          PoetUtil.createGroupedPropertiesGetter(
+              entry.getKey(),
+              entry.getValue().stream().map(c -> c.getName()).toList(),
+              propertyType);
+      methodSpecs.add(methodSpec);
+    }
+    return methodSpecs;
+  }
+
   /**
    * Data class holding all of the information required to create accessor methods (setter/getter)
    * for a field.
    */
   @Data
-  @AllArgsConstructor
+  @RequiredArgsConstructor
```

Comment:
This class is meant to be immutable.  All fields need to be final.  I think it would be better to add a `@Builder` annotation and change constructor calls to use the builder going forward.  The builder will default any uninitialized object references to null.